securedrop/i18n_tool.py

```diff
@@ -57,196 +59,371 @@ def sh(command, input=None):
+        parser.add_argument(
+            '--compile',
+            action='store_true',
+            help='compile translations')
+        parser.add_argument(
+            '--translations-dir',
+            default=translations_dir,
+            help='Base directory for translation files (default {})'.format(
+                translations_dir))
+        parser.add_argument(
+            '--version',
+            default=version.__version__,
+            help=('SecureDrop version '
+                  'to store in pot files (default {})'.format(
+                      version.__version__)))
+        parser.add_argument(
+            '--sources',
+            default=sources,
+            help='Source files and directories to extract (default {})'.format(
+                sources))
+
+    def set_translate_messages_parser(self, subps):
+        parser = subps.add_parser('translate-messages',
+                                  help=('Update and compile '
+                                        'source and template translations'))
+        translations_dir = join(dirname(realpath(__file__)), 'translations')
+        sources = '.,source_templates,journalist_templates'
+        self.set_translate_parser(subps, parser, translations_dir, sources)
+        mapping = 'babel.cfg'
+        parser.add_argument(
+            '--mapping',
+            default=mapping,
+            help='Mapping of files to consider (default {})'.format(
+                mapping))
+        parser.set_defaults(func=self.translate_messages)
+
+    def set_translate_desktop_parser(self, subps):
+        parser = subps.add_parser('translate-desktop',
+                                  help=('Update and compile '
+                                        'desktop icons translations'))
+        translations_dir = join(
+            dirname(realpath(__file__)),
+            '../install_files/ansible-base/roles/tails-config/templates')
+        sources = 'desktop-journalist-icon.j2.in,desktop-source-icon.j2.in'
+        self.set_translate_parser(subps, parser, translations_dir, sources)
+        parser.set_defaults(func=self.translate_desktop)
+
+    @staticmethod
+    def require_git_email_name(git_dir):
+        cmd = ('git -C {d} config --get user.name > /dev/null && '
+               'git -C {d} config --get user.email > /dev/null'.format(
+                   d=git_dir))
+        if subprocess.call(cmd, shell=True):  # nosec
+            raise Exception(cmd + ' returned false, please set name and email')
+        return True
+
+    def update_docs(self, args):
+        l10n_content = '.. GENERATED BY i18n_tool.py DO NOT EDIT:\n\n'
+        for (code, info) in sorted(I18NTool.SUPPORTED_LANGUAGES.items()):
+            l10n_content += '* ' + info['name'] + ' (' + code + ')\n'
+        includes = join(args.documentation_dir, 'includes')
+        l10n_txt = join(includes, 'l10n.txt')
+        open(l10n_txt, 'w').write(l10n_content)
+        self.require_git_email_name(includes)
```

Comment:
Is `includes` the right directory to be passed as `git_dir` to `require_git_email_name`? I'm getting an exception when I try to run this utility, even with my name and email defined in a repo-level `.gitconfig`.

Reply:
it does not matter if the command is not run at the root of the git repository, the important thing is that it runs within a git repository